crates/sui-graphql-rpc/tests/dot_move_e2e.rs

```diff
@@ -0,0 +1,445 @@
+        test_infra::cluster::{
+            start_graphql_server_with_fn_rpc, start_network_cluster, wait_for_graphql_server,
+            NetworkCluster,
+        },
+    };
+    use sui_graphql_rpc_client::simple_client::SimpleClient;
+    use sui_json_rpc_types::ObjectChange;
+    use sui_move_build::BuildConfig;
+    use sui_types::{
+        base_types::{ObjectID, SequenceNumber},
+        digests::ObjectDigest,
+        move_package::UpgradePolicy,
+        object::Owner,
+        programmable_transaction_builder::ProgrammableTransactionBuilder,
+        transaction::{CallArg, ObjectArg},
+        Identifier, SUI_FRAMEWORK_PACKAGE_ID,
+    };
+    use tokio::time::sleep;
+
+    const DOT_MOVE_PKG: &str = "tests/dot_move/dot_move/";
+    const DEMO_PKG: &str = "tests/dot_move/demo/";
+    const DEMO_PKG_V2: &str = "tests/dot_move/demo_v2/";
+    const DEMO_PKG_V3: &str = "tests/dot_move/demo_v3/";
+
+    #[derive(Clone, Debug)]
+    struct UpgradeCap(ObjectID, SequenceNumber, ObjectDigest);
+
+    #[tokio::test]
+    async fn test_dot_move_e2e() {
+        let network_cluster =
+            start_network_cluster(ConnectionConfig::ci_integration_test_cfg(), None).await;
+
+        let external_network_chain_id = network_cluster
+            .validator_fullnode_handle
+            .fullnode_handle
+            .sui_client
+            .read_api()
+            .get_chain_identifier()
+            .await
+            .unwrap();
+
+        eprintln!("External chain id: {:?}", external_network_chain_id);
+
+        // publish the dot move package in the internal resolution cluster.
+        let (pkg_id, registry_id) = publish_dot_move_package(&network_cluster).await;
+
+        let (v1, v2, v3) = publish_demo_pkg(&network_cluster).await;
+
+        let name = "app@org".to_string();
+
+        // Register the package: First, for the "base" chain state.
+        register_pkg(
+            &network_cluster,
+            pkg_id,
+            registry_id,
+            v1,
+            name.clone(),
+            None,
+        )
+        .await;
+
+        // Register the package for the external resolver.
+        register_pkg(
+            &network_cluster,
+            pkg_id,
+            registry_id,
+            v1,
+            name.clone(),
+            Some(external_network_chain_id.clone()),
+        )
+        .await;
+
+        // Initialize the internal and external clients of GraphQL.
+
+        // The first cluster uses internal resolution (mimics mainnet, does not rely on external chain).
+        let internal_client = init_dot_move_gql(
+            8000,
+            9184,
+            ServiceConfig::dot_move_test_defaults(
+                false,
+                None,
+                Some(pkg_id.into()),
+                Some(registry_id.0),
+                None,
+            ),
+        )
+        .await;
+
+        // Wait for the transactions to be committed and indexed
+        sleep(Duration::from_secs(5)).await;
```

Comment:
We should have a `wait_for_checkpoint` helper or similar which may work better here? Not sure if you're running your tests in simulated mode or not (which allows you to control when checkpoints happen, and therefore also when indexing happens).

Reply:
Yeah... I took the lazy approach, but I now changed it to query the latest checkpoint from the FN and wait until GQL is caught up to this.